File: MATLAB/fizzbuzz.m
%{
Run with MATLAB
%}

function [] = fizzbuzz()

    % Loop
    function [] = main()
        maxNum = input("Please enter max number: ");
        for i = 1:maxNum
            disp(divisible(i))
        end
    end
    
    % Logic
    function string = divisible(i)
        if mod(i,3) == 0 && mod(i,5) == 0
            string = "fizzBuzz";
            return
        elseif mod(i,3) == 0
            string = "fizz";
            return
        elseif mod(i,5) == 0
            string = "Buzz";
            return
        else
            string = int2str(i);
        end
    end

    % Main
    main()
end

File: Rust/fizzbuzz.rs
/*
Compile with rustc fizzbuzz.rs
Run with ./fizzbuzz
*/

use std::io::Write;

// Logic
fn divisible(i:u32) -> String {
    if i%3 == 0 && i%5 == 0 {
        return "fizzBuzz".to_string();
    } else if i%3 == 0 {
        return "fizz".to_string();
    } else if i%5 == 0 {
        return "Buzz".to_string();
    } else {
        return i.to_string();
    }
}

// Main
fn main() {
    let mut input = String::new();
    print!("Please enter max number: ");
    std::io::stdout().flush().unwrap();
    std::io::stdin().read_line(&mut input).unwrap();
    let max_num = input.trim_end().parse::<u32>().unwrap();
    // Loop
    for i in 1..max_num+1 {
        println!("{}", divisible(i));
    }
}
